### PR TITLE
Add module Store, StoreUnit & Add relation to Product

### PR DESCRIPTION
**Pull request by Haruto000**
-Add module with scaffold:module
 -Add module enitity with scaffold:entity Store Store
 -Add Table -> create_stores_table
 -Migrate Table with php artisan module:migrate Store

### CODE REVIEW

File: Modules/Product/Resources/views/admin/products/tabs/inventory.blade.php [hunk 1]
```diff
@@ -1,12 +1,37 @@
...
+                <div class="col-sm-5">
+
+                    @php
+                    $pivottable = $storeunit->Products()->where("product_id", $product->id)->withPivot("quantity", "in_stock")->first();
+                    $pivottable = $pivottable->pivot ?? null;
+                    @endphp
+
+            	{{ Form::number('quantity[]', $store->name."/".$storeunit->name, $errors, $pivottable, ['required' => true] ) }}
+
+            	<input type="hidden" name="unit[]" value="{{$storeunit->id}}">
```
Comment (ankitit2015) [inline Modules/Product/Resources/views/admin/products/tabs/inventory.blade.php:22]: Use proper indentation in this file

File: Modules/Product/Resources/views/admin/products/tabs/inventory.blade.php [hunk 2]
```diff
@@ -1,12 +1,37 @@
...
+                </div>
+
+                <div class="col-sm-5">
+
+                {{ Form::select('in_stock[]', 'Availability', $errors, ['In Stock' ,'Out Of Stock'] ,$pivottable, trans('product::products.form.stock_availability_states')) }}
```
Comment (ankitit2015) [inline Modules/Product/Resources/views/admin/products/tabs/inventory.blade.php:27]: ['In Stock' ,'Out Of Stock']   Pass this from controller, That too translated strings

File: Modules/Store/Admin/StoreTabs.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+use Modules\Admin\Ui\Tabs;
+use Modules\Store\Entities\StoreUnit;
+use Modules\Store\Entities\Store;
+
+class StoreTabs extends Tabs
+{
+
+    public function make()
+    {
+        $this->group('store_set_information', trans('store::admin.tabs.group.store_information'))
```
Comment (ankitit2015) [inline Modules/Store/Admin/StoreTabs.php:15]: 'store_set_information' => 'store_information'

File: Modules/Store/Admin/StoreTabs.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            ->active()
+            ->add($this->general());
+    }
+
+    private function general()
+    {
+        return tap(new Tab('Add StoreUnit', trans('store::admin.tabs.general')), function (Tab $tab) {
```
Comment (ankitit2015) [inline Modules/Store/Admin/StoreTabs.php:22]: 'Add StoreUnit' => 'Add Store Unit'

File: Modules/Store/Admin/StoreTabs.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            $tab->active();
+            $tab->weight(5);
+            $tab->fields(['name']);
+            $tab->view('store::admin.stores.tabs.addstore', [
```
Comment (ankitit2015) [inline Modules/Store/Admin/StoreTabs.php:26]: 'store::admin.stores.tabs.addstore' => 'store::admin.stores.tabs.addStore'

File: Modules/Store/Admin/StoreTabs.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+                'attributeSets' => $this->getAttributeSets(),
+            ]);
+        });
+    }
+
+    private function getAttributeSets()
+    {
+        $attributeSets = Store::all()->sortBy('name')->pluck('name', 'id');
```
Comment (ankitit2015) [inline Modules/Store/Admin/StoreTabs.php:34]: $attributeSets =>$stores

File: Modules/Store/Admin/StoreTabs.php [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            $tab->active();
+            $tab->weight(5);
+            $tab->fields(['name']);
+            $tab->view('store::admin.stores.tabs.addstore', [
+                'attributeSets' => $this->getAttributeSets(),
+            ]);
+        });
+    }
+
+    private function getAttributeSets()
```
Comment (ankitit2015) [inline Modules/Store/Admin/StoreTabs.php:32]: getStores()

File: Modules/Store/Config/assets.php [hunk 8]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+return [
+    /*
+    |--------------------------------------------------------------------------
+    | Define which assets will be available through the asset manager
+    |--------------------------------------------------------------------------
+    | These assets are registered on the asset manager
+    */
+    'all-assets' => [
```
Comment (ankitit2015) [inline Modules/Store/Config/assets.php:10]: 'all-assets' => 'all_assets'
 'required-assets' =>  'required_assets'

File: Modules/Store/Resources/lang/en/admin.php [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+        'group' => [
+            'store_information' => 'Store Information',
+        ],
+        'general' => 'Add Store',
+    ],
+
+    'table' => [
+    	'id' => 'Store Id',
+    	'name' => 'Store Name',
+    	'latitude_longitude' => 'Store Latitude/Longitude',
```
Comment (ankitit2015) [inline Modules/Store/Resources/lang/en/admin.php:15]: 'Store Latitude/Longitude' => 'Store Latitude & Longitude'

File: Modules/Store/Resources/lang/en/attributes.php [hunk 10]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+return [
+
+	'form' => [
+		'name' => 'Name',
+		'latitude_longitude' => 'Latitude/Longitude',
```
Comment (ankitit2015) [inline Modules/Store/Resources/lang/en/attributes.php:7]: Latitude & Longitude

File: Modules/Store/Resources/views/admin/stores/create.blade.php [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    @slot('title', trans('admin::resource.create', ['resource' => trans('store::stores.store')]))
+
+    <li><a href="{{ route('admin.stores.index') }}">{{ trans('store::stores.stores') }}</a></li>
+    <li class="active">{{ trans('admin::resource.create', ['resource' => trans('store::stores.store')]) }}</li>
+@endcomponent
+
+@section('content')
+    
+    <form method="POST" action="{{ route('admin.stores.store') }}" class="form-horizontal" id="store-create-form" novalidate>
+
```
Comment (ankitit2015) [inline Modules/Store/Resources/views/admin/stores/create.blade.php:14]: Remove new line

File: Modules/Store/Resources/views/admin/stores/create.blade.php [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+        {{ csrf_field() }}
```
Comment (ankitit2015) [inline Modules/Store/Resources/views/admin/stores/create.blade.php:15]: Proper indentation

File: Modules/Store/Resources/views/admin/stores/create.blade.php [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+
```
Comment (ankitit2015) [inline Modules/Store/Resources/views/admin/stores/create.blade.php:16]: Remove new line

File: Modules/StoreUnit/Resources/views/admin/storeunits/tabs/templates/tab.blade.php [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+                
+                <div class='tab-pane fade in active' id='general'><h3 class='tab-content-title'>
+                {{ trans('store::stores.createstore.addstore') }}</h3><div class="row">
+            <div class="col-md-8">
+                @include('store::admin.stores.tabs.form.addstore')
+            </div>
+        </div>
+
+
+    </div><div class='tab-pane fade in ' id='values'><h3 class='tab-content-title'>{{ trans('store::stores.unitstore.storeunit') }}</h3><div id="attribute-values-wrapper">
```
Comment (ankitit2015) [inline Modules/StoreUnit/Resources/views/admin/storeunits/tabs/templates/tab.blade.php:52]: Not understainding this translation 'store::stores.unitstore.storeunit'

File: Modules/StoreUnit/Routes/admin.php [hunk 15]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+Route::get('storeunits', [
```
Comment (ankitit2015) [inline Modules/StoreUnit/Routes/admin.php:3]: 'storeunits'=>'storeUnits'

File: Modules/StoreUnit/Sidebar/SidebarExtender.php [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+use Maatwebsite\Sidebar\Group;
+use Modules\Admin\Sidebar\BaseSidebarExtender;
+
+
+class SidebarExtender extends BaseSidebarExtender
+{
+
+    public function extend(Menu $menu)
+    {
+
```
Comment (ankitit2015) [inline Modules/StoreUnit/Sidebar/SidebarExtender.php:16]: Remove unnecesary new lines

File: Modules/StoreUnit/composer.json [hunk 17]
```diff
@@ -0,0 +1,28 @@
+{
+    "name": "fleetcart/storeunit",
```
Comment (ankitit2015) [inline Modules/StoreUnit/composer.json:2]: storeUnit

File: app/Product.php [hunk 18]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace FleetCart;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Product extends Model
```
Comment (ankitit2015) [inline app/Product.php:7]: Reason for adding this ?

File: app/Console/Commands/ScaffoldModuleCommand.php [hunk 19]
```diff
@@ -12,20 +12,23 @@ class ScaffoldModuleCommand extends Command
...
+    
     protected $name = 'scaffold:module';
 
     /**
      * The console command description.
      *
      * @var string
-     */
+    */
+
```
Comment (ankitit2015) [inline app/Console/Commands/ScaffoldModuleCommand.php:23]: remove unnecessary changes to this file

File: storage/framework/views/065c9e36e7e54ee5e2d712b5ee29f57d778c709c.php [hunk 20]
```diff
@@ -0,0 +1,39 @@
+<div class="modal fade" id="confirmation-modal" tabindex="-1" role="dialog" aria-hidden="true">
```
Comment (ankitit2015) [inline storage/framework/views/065c9e36e7e54ee5e2d712b5ee29f57d778c709c.php:1]: push changes after clearing views, these are cached views

File: Modules/Product/Resources/views/admin/products/tabs/inventory.blade.php [hunk 21]
```diff
@@ -1,12 +1,37 @@
...
+
+    	
+        
         {{ Form::text('sku', trans('product::attributes.sku'), $errors, $product) }}
         {{ Form::select('manage_stock', trans('product::attributes.manage_stock'), $errors, trans('product::products.form.manage_stock_states'), $product) }}
 
-        <div class="{{ old('manage_stock', $product->manage_stock) ? '' : 'hide' }}" id="qty-field">
-            {{ Form::number('qty', trans('product::attributes.qty'), $errors, $product, ['required' => true]) }}
-        </div>
+            @foreach((Modules\Store\Entities\Store::all() ?? []) as $store)
```
[the remark this answers is not in the document]
Comment (Haruto000) [reply Modules/Product/Resources/views/admin/products/tabs/inventory.blade.php:9]: i tried !

File: Modules/Store/Admin/StoreTabs.php (hunk 5 above)
Comment (Haruto000) [reply Modules/Store/Admin/StoreTabs.php:26]: why ?
